### PR TITLE
[Sprint #5] developing basic APIs

### PR DESCRIPTION
**Pull request by dobestan**
#### Summary
- basic retrieve APIs ( how about Create / Update / Destroy ? it has been intentionally excluded from current API because we don't have any model level validation, currently using django's manager level validation. )
- APIs for search : will be used for auto-complete and search feature in client development.
#### Developed APIs
- `/api/users/`
- `/api/users/?search={query}`
- `/api/projects/`
- `/api/projects/?search={query}`
- `/api/teams/`
- `/api/teams/?search={query}`
- `/api/teams/<teamname>/users/`
- `/api/teams/<teamname>/users/?search={query}`
- `/api/teams/<teamname>/projects/`
- `/api/teams/<teamname>/projects/?search={query}`

### CODE REVIEW

**Comment by BOBTommy** [comment]
I think we need to CRUD all function for REST, you will update? or add?

File: buildbuild/teams/models.py [hunk 1]
```diff
@@ -101,4 +101,4 @@ class WaitList(models.Model):
     user = models.ForeignKey('users.User', related_name="+")
     request_date = models.DateTimeField(default=timezone.now())
```
Comment (BOBTommy) [inline buildbuild/teams/models.py]: This field also change default=timezone.now() -> auto_now_add = True
It will be good.

**Comment by dobestan** [comment]
@BOBTommy : i have mentioned that on `Summary`

> how about Create / Update / Destroy ? it has been intentionally excluded from current API because we don't have any model level validation, currently using django's manager level validation.

in our current model implementation, every field validated on Manager layer, ( not Model layer ) so we need to add Model level validation to update API later. Thanks.

and actually we have to talk about `Create`, `Update`, and 'Destroy` features are really needed for our service. ( alternatively, we can just use django's basic ModelForm instead. )

File: buildbuild/teams/models.py (hunk 1 above)
Comment (dobestan) [inline buildbuild/teams/models.py]: Thank for your advice.
I will update this right away~

**Pull request merged**